portfolio/src/main/java/com/google/sps/servlets/Delete.java

```diff
@@ -0,0 +1,43 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.KeyRange;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;;
+import java.io.IOException;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/delete")
+public class Delete extends HttpServlet {
+
+  private Gson gson = new Gson();
+
+  /** Delete the element of the given key id */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String reader = request.getReader().lines().collect(Collectors.joining());
+    JsonObject deleteObj = new JsonParser().parse(reader).getAsJsonObject();
+    long id = Long.parseLong(deleteObj.get("id").getAsString());
+    String type = deleteObj.get("type").getAsString();
+
+    Query query = new Query(type).addSort("timeStamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Key deleteKey = KeyFactory.createKey(type, id);
```

Comment:
I'm not familiar with how keys are determined in the datastoreservice, but is it that this gives you the key of the entry of given type and id? 

also if each id is unique, why do you need to specify the type? 
is it not enough to use `entity.getKey()`?

Reply:
I also didnt know how it worked much so i went onto Cupcake for help.
I could try tesing out taking out the type and see if the unique ID will still work.

Reply:
yea it doesnt work without it I get a compile error, im not sure how Keys work much either

Reply:
ok if it's functionally correct then resolve this and i'll approve it.